metadata/metadata_test.go

```diff
@@ -555,6 +555,126 @@ var _ = Describe("Metadata", func() {
 		})
 	})
 
+	When("SetAnnotation is called", func() {
+
+		It("should error if the object is nil", func() {
+			err := SetAnnotation(nil, "foo", "bar")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("object cannot be nil"))
+		})
+
+		It("should add the annotation to the object", func() {
+			annotations := map[string]string{"foo": "bar"}
+			pod := &corev1.Pod{}
+
+			Expect(pod.Annotations).To(HaveLen(0))
+			for key, value := range annotations {
+				Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			}
+			for key, value := range annotations {
+				Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+			}
```

Comment:
What about setting more annotations and then compare the maps? Something like this:
```
                        annotations := map[string]string{"foo": "bar", "baz": "qux"}
			pod := &corev1.Pod{}

			Expect(pod.Annotations).To(HaveLen(0))

			for key, value := range annotations {
				Expect(SetAnnotation(pod, key, value)).To(Succeed())
			}

			Expect(pod.Annotations).To(Equal(annotations))
```

Reply:
How exactly will this improve test coverage?

Reply:
In this case it will ensure that there are no extra annotations after running. But it also removes the second loop.

Reply:
I used this with single annotation
```
Expect(pod.Annotations).To(Equal(annotations))
```